concordia/models.py

```diff
@@ -84,16 +84,41 @@ def get_absolute_url(self):
         return reverse("transcriptions:campaign-detail", args=(self.slug,))
 
 
+class Topic(models.Model):
+    objects = PublicationQuerySet.as_manager()
+
+    published = models.BooleanField(default=False, blank=True)
+
+    ordering = models.IntegerField(
+        default=0, help_text="Sort order override: lower values will be listed first"
+    )
+    title = models.CharField(blank=False, max_length=255)
+    slug = models.SlugField(blank=False, allow_unicode=True, max_length=80)
+    description = models.TextField(blank=True)
+    thumbnail_image = models.ImageField(
+        upload_to="topic-thumbnails", blank=True, null=True
+    )
+    short_description = models.TextField(blank=True)
+
+    class Meta:
+        ordering = ["title"]
+
+    def __str__(self):
+        return self.title
+
+    def get_absolute_url(self):
+        return reverse("topic-detail", kwargs={"slug": self.slug})
+
+
 class Resource(MetricsModelMixin("resource"), models.Model):
     sequence = models.PositiveIntegerField(default=1)
     title = models.CharField(blank=False, max_length=255)
     resource_url = models.URLField()
 
-    campaign = models.ForeignKey(Campaign, on_delete=models.CASCADE)
-
-    class Meta:
```

Comment:
Was it intentional to remove the unique/ordering test? The history wasn't clear to me from the commit & I was wondering whether we might have wanted to add topic into the check, enforce it in a clean method, etc.

Reply:
It's intentional. Either campaign or topic can be specified as a foreign key so they may also be blank.

Reply:
Should we make those relationships nullable so we can still have the uniqueness constraints on non-null values?